test/Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests/DiagnosticTest.cs

```diff
@@ -0,0 +1,105 @@
+using System.Text;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests
+{
+    public class DiagnosticTest
+    {
+        [Fact]
+        public void SingleActivityProvider_Should_Create_Activity()
+        {
+            var consumerGroup = "my-consumer-group";
+            var numActivityTags = 7;
+            var traceId = ActivityTraceId.CreateRandom();
+            var spanId = ActivitySpanId.CreateRandom();
+            string traceparent = "00-" + traceId + "-" + spanId + "-" + "01";
+            var kafkaEvent = new KafkaEventData<string, string>("key", "value");
+            kafkaEvent.Headers.Add("traceparent", Encoding.ASCII.GetBytes(traceparent));
+            kafkaEvent.Topic = "mytopic";
+            kafkaEvent.Partition = 1;
+
+            var activityListener = new ActivityListener
+            {
+                ShouldListenTo = _ => true,
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData,
+            };
+            ActivitySource.AddActivityListener(activityListener);
+            var singleEventActivityProvider = new SingleEventActivityProvider(kafkaEvent, consumerGroup);
+            Assert.NotNull(singleEventActivityProvider.Activity);
+            var activity = singleEventActivityProvider.Activity;
+            Assert.Equal(traceId, activity.TraceId);
+            Assert.Equal(numActivityTags, activity.Tags.Count());
+            Assert.Equal("kafka", activity.GetTagItem(ActivityTags.System).ToString());
+            Assert.Equal(kafkaEvent.Topic, activity.GetTagItem(ActivityTags.DestinationName).ToString());
+            Assert.Equal("topic", activity.GetTagItem(ActivityTags.DestinationKind).ToString());
+            Assert.Equal("process", activity.GetTagItem(ActivityTags.Operation).ToString());
+            Assert.Equal(consumerGroup, activity.GetTagItem(ActivityTags.KafkaConsumerGroup).ToString());
+            Assert.Equal(kafkaEvent.Partition.ToString(), activity.GetTagItem(ActivityTags.KafkaPartition));
+            var key = activity.GetTagItem(ActivityTags.KafkaMessageKey).ToString();
+            Assert.Equal(kafkaEvent.Key, activity.GetTagItem(ActivityTags.KafkaMessageKey).ToString());
+            singleEventActivityProvider.Dispose();
+            activityListener.Dispose();
+        }
+
+        [Fact]
+        public void BatchActivityProvider_Should_Create_Activity_With_Links()
+        {
+            var consumerGroup = "my-consumer-group";
+            int numkafkaEvents = 10;
+            var kafkaEvents = new KafkaEventData<string, string>[numkafkaEvents];
+            var topicName = "mytopic";
+            var numActivityTags = 5;
+            for (int i = 0; i< numkafkaEvents; i++)
+            {
+                var traceId = ActivityTraceId.CreateRandom();
+                var spanId = ActivitySpanId.CreateRandom();
+                string traceparent = "00-" + traceId + "-" + spanId + "-" + "01";
```

Comment:
this is being used in multiple places and looks a constant, can we try with a function and passing traceid & spanid and return the traceparent id ?

Reply:
Done